src/main/java/org/apache/maven/plugins/gpg/AbstractGpgMojo.java

```diff
@@ -299,11 +300,18 @@ public final void execute() throws MojoExecutionException, MojoFailureException
             // We're skipping the signing stuff
             return;
         }
-        if (bestPractices && (isNotBlank(passphrase) || isNotBlank(passphraseServerId))) {
-            // Stop propagating worst practices: passphrase MUST NOT be in any file on disk
-            throw new MojoFailureException(
-                    "Do not store passphrase in any file (disk or SCM repository), rely on GnuPG agent or provide passphrase in "
-                            + passphraseEnvName + " environment variable.");
+        if (bestPractices) {
+            if (isNotBlank(passphrase) || isNotBlank(passphraseServerId)) {
+                // Stop propagating worst practices: passphrase MUST NOT be in any file on disk
+                throw new MojoFailureException(
+                        "Do not store passphrase in any file (disk or SCM repository), rely on GnuPG agent or provide passphrase in "
+                                + passphraseEnvName + " environment variable.");
+            }
+        } else {
+            if (!isNotBlank(passphraseServerId)) {
+                // default it programmatically: this is needed to handle different cases re bestPractices
+                passphraseServerId = GPG_PASSPHRASE;
+            }
```

Comment:
I feel this is a default case inside an else-if nesting. However, as the isolated if(bestPractices) improves readability, I'd say: keep it for now.

One solution would be to refactor out a method like `setDefaults()`, but probably not worth it.

Reply:
These fields are deprecated and are meant to be removed.. so IMO that would be overkill.